lib/parsers.js

```diff
@@ -58,6 +60,15 @@ function getContextFromRequest (req, conf, type) {
       Object.assign({}, req.headers),
       conf.sanitizeFieldNamesRegExp
     )
+
+    // TODO: deprecate filterHttpHeaders in next major release
```

Comment:
We might be disagreeing on the definition of "deprecate". For me, "deprecating" means that we would document (and mark in types) that the config var is "deprecated" -- hence users should avoid it and the docs should show users how to migrate away from it, but the field is still supported.

The "TODO" for the next major release would be to *remove* the `filterHttpHeaders` config var.
Thoughts?

If we agree, then would you like to do the deprecation tweaks in this PR, or prefer that we do it in a separate change?

Reply:
I think your approach is much better. I shall do the changes in this PR.

Thanks!